grr/server/grr_response_server/output_plugins/elasticsearch_plugin.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+# Lint as: python3
+"""OutputPlugin that sends Flow results to an Elasticsearch cluster.
+
+Configuration values for this plugin can be found in
+core/grr_response_core/config/output_plugins.py
+
+The specification for the indexing of documents is
+https://www.elastic.co/guide/en/elasticsearch/reference/7.1/docs-index_.html
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import unicode_literals
+
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Text
+from urllib import parse as urlparse
+
+import requests
+
+from google.protobuf import json_format
+from grr_response_core import config
+from grr_response_core.lib.rdfvalues import flows as rdf_flows
+from grr_response_core.lib.rdfvalues import protodict as rdf_protodict
+from grr_response_core.lib.rdfvalues import structs as rdf_structs
+from grr_response_core.lib.util.compat import json
+from grr_response_proto import output_plugin_pb2
+from grr_response_server import data_store
+from grr_response_server import export
+from grr_response_server import output_plugin
+from grr_response_server.gui.api_plugins import flow as api_flow
+
+BULK_OPERATIONS_PATH = "_bulk"
+
+JsonDict = Dict[Text, Any]
```

Comment:
Could you add a `TODO` comment linking to the [JSON type discussion](https://github.com/python/typing/issues/182)? Once they figure it out, we should fix this one day...